qt/widgets/plotting/src/Mpl/ExternalPlotter.cpp

```diff
@@ -130,19 +139,22 @@ void ExternalPlotter::plotSpectra(std::string const &workspaceName, std::string
  *
  * @param workspaceNames List of names of workspaces to plot
  * @param workspaceIndices List of indices to plot
+ * @param errorBars List of booleans to add/remove error bars to each line individually
  */
 void ExternalPlotter::plotCorrespondingSpectra(std::vector<std::string> const &workspaceNames,
-                                               std::vector<int> const &workspaceIndices, bool errorBars) {
-  if (workspaceNames.empty() || workspaceIndices.empty())
+                                               std::vector<int> const &workspaceIndices, std::vector<bool> errorBars,
+                                               std::vector<boost::optional<QHash<QString, QVariant>>> kwargs) {
+  if (workspaceNames.empty() || workspaceIndices.empty() || errorBars.empty())
     return;
-  if (workspaceNames.size() > 1 && workspaceNames.size() != workspaceIndices.size())
+  if (workspaceNames.size() > 1 && workspaceNames.size() != workspaceIndices.size() &&
+      workspaceNames.size() != errorBars.size())
```

Comment:
I think we should also check the `kwargs` size, and some of the size comparisons should be `||` rather than `&&`
```suggestion
  auto const numberOfPlots = workspaceNames.size();
  if (numberOfPlots > 1 && (workspaceIndices.size() != numberOfPlots || errorBars.size() != numberOfPlots || kwargs.size() != numberOfPlots))
```

Reply:
Was there a reason for checking if the number of plots is greater than 1?